Sources/RemoteMessaging/RemoteMessagingStore.swift

```diff
@@ -47,45 +47,45 @@ public final class RemoteMessagingStore: RemoteMessagingStoring {
         public static let privateContextName = "RemoteMessaging"
     }
 
-    let context: NSManagedObjectContext
+    let database: CoreDataDatabase
```

Comment:
This change makes the functions in RemoteMessagingStore create context as needed, ensuring that they have the fresh data all the time. We use a similar approach in Sync Bookmarks provider.